### ISSUE TITLE
Double file extention during share on Android

### ISSUE THREAD

**Issue by remixoff** [nextcloud/android#1882]
### Actual behaviour
Files in cloud have *.JPG.jpg mask. Original files not.

### Expected behaviour
Doesn't change filenames during upload
 
### Steps to reproduce
1. Open QuickPic
2. Go to the Camera folder
3. Click share->Nextcloud->Upload. 

### Environment data
Android version:

Device model: 

Stock or customized system:

Nextcloud app version:

Nextcloud server version:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by AndyScherzinger** [comment]
@remixoff I just installed QuickPic and shared an images which I named Test.JPG and did 

* Open QuickPic
* Go to the Camera folder
* Click share->Nextcloud->Upload.

The file didn't get any filename/extension change and uploaded "as-is". @tobiasKaminsky do you have any idea? Afaik we do not change file names/extensions at all except when we need to created "duplicates" due to file name collisions which to me doesn't seem to be the scenario here.

**Comment by remixoff** [comment]
Might be the reason is capital letters in original ext? I have shared screenshots here: http://cloud.remixoff.ru/index.php/s/bkdM8Zs08tsQ3Xu

sorry, NC does not allow to make scrshots, I've attached the log from svr. Sorry, I am writing from mobile, didn't hightlight log line.

**Comment by remixoff** [comment]
btw, at the upload page (can't be shoted by NC reasons) I already see double ext.

**Comment by tobiasKaminsky** [comment]
NC app is not doing anything to the suffix. It only appends (2), (3)... if file name collision is detected.
Also it is weird that you say you cannot take a screenshot from NC app as we do not block this.

As you say that it is already wrong in the upload pane, where you select the file from your device, this leads to the impression that it is already wrong on the device.
Can you verify this with a file browser?

**Comment by tobiasKaminsky** [comment]
![2017-12-20-135844](https://user-images.githubusercontent.com/5836855/34208239-f48b30c0-e58d-11e7-85a0-30747d954c85.png) ![2017-12-20-135858](https://user-images.githubusercontent.com/5836855/34208238-f46fc4a2-e58d-11e7-9fd6-e7a3bc2cc470.png)

This works for me without any renaming.

**Comment by AndyScherzinger** [comment]
> Might be the reason is capital letters in original ext?

I also tested it with .JPG and it got uploaded with its original file name and extension.

**Comment by remixoff** [comment]
Please, test with filename DSC_9505.JPG. I just renamed file to alex.jpg. It was uploaded correctly. Checking if the underscore is the reason. Will comment in few minutes

**Comment by remixoff** [comment]
al_ex.jpg is ok
alex_0.jpg is ok
alex_9998.jpg is ok
DSC_9997.JPG - wrong
DSCa_9997.JPG - wrong
dsc_9997.JPG - wrong
desc_9997.JPG - wrong
alex_9997.JPG - wrong

well.... it affected only files under mask "*_NNNN.JPG", where N - number. 
Magic? )

**Comment by tobiasKaminsky** [comment]
Thank you very much for the intensive bug hunting.
I used gallery/amaze for sharing the image as I hadn't installed quickpic and I could not reproduce the issue.
But with quickpic I can reproduce the issue. So something quickpic is doing "wrong"/in another way.

**Comment by AndyScherzinger** [comment]
merged for 2.1 so it will ship to the dev build tonight 🎉
